src/main/java/org/folio/circulation/services/FeeFineFacade.java

```diff
@@ -150,13 +147,33 @@ public FeeFineFacade(Clients clients) {
     return feeFineActionRepository.createAll(context.getActions());
   }
 
+  private CompletableFuture<Result<StoredFeeFineActionBuilder>> populateCreatedBy(
+    StoredFeeFineActionBuilder builder, CreateAccountCommand command) {
+
+    if (command.isCreatedByAutomatedProcess()) {
+      return completedFuture(succeeded(builder.createdByAutomatedProcess()));
+    }
+
+    return userRepository.getUser(command.getStaffUserId())
+      .thenApply(r -> r.map(builder::withCreatedBy));
+  }
+
   private CompletableFuture<Result<User>> fetchUser(String userId) {
     return userRepository.getUser(userId);
   }
 
   private CompletableFuture<Result<ServicePoint>> fetchServicePoint(String servicePointId) {
-    return servicePointRepository.getServicePointById(servicePointId)
-      // To prevent NPE.
-      .thenApply(r -> r.map(sp -> sp == null ? new ServicePoint(new JsonObject()) : sp));
+    return servicePointRepository.getServicePointById(servicePointId);
```

Comment:
Does the repository handle the null pointer situation that the comment refers to?

Reply:
This NPE handling logic has moved to StoredFeeFineAction class, that represents a fee/fine action.